### ISSUE TITLE
Consider moving to remorhaz/php-json-path

### ISSUE THREAD

**Issue by dafeder** [GetDKAN/RootedJsonData#5]
Appears to be more robust, and according to the [JSONPath Comparison site](https://cburgmer.github.io/json-path-comparison/) is actually more compliant than the @Galbar version. The API is more complex however and would require more code in our class to support the same basic get/set functionality.

Mostly creating this as an icebox issue that can be resurrected if we actually run into issues or limitations with https://github.com/Galbar/JsonPath-PHP

**Comment by dafeder** [comment]
Update: galbar has been working fine for our purposes, though remorhaz/php-json-path continues to be more actively maintained.

**Comment by Galbar** [comment]
I have not abandoned Galbar/JsonPath-PHP. It just feels feature complete to me and thus I've felt no need to update it. Looking at remorhaz/php-json-path, I considered at one point the possibility of pre-building queries but the library has been working just fine as it is so I never got to doing it.

(I just got a notification. I thought I'd confirm that the library is not abandoned 😉 )

**Comment by dafeder** [comment]
Thanks @Galbar for the clarification! I think it's doubtful we'll move at this point, and I may have sounded harsher there than I meant to; a year is not that long for such a feature-complete library! Mostly concerned about continued compatibility with new versions of PHP - tests appear to run only on 5.4?

**Comment by Galbar** [comment]
You are correct. Tests run only on 5.4. I did look to make them run in newer versions but there was some compatibility issues with at least one dependency (I believe it was the testing framework) and at the time I was too busy to fix this. I'll create an issue to update the tests for php 7 and 8. I can't promise I'll be able to tackle it soon, though.

On the other hand, if you are met with any bugs or compatibility issues let me know (by reporting an issue in the github repo) and I'll address it promptly.

**Comment by dafeder** [comment]
Will do. I might attempt a PR as we've done a lot of this kind of stuff on our end lately, but similarly can't guarantee 😄

**Comment by TheDigitalOrchard** [comment]
Just FYI, I'm running this library (not the tests) on PHP 8.3 without any issues, so that's positive.
